src/database-controller/src/common/framework.js

```diff
@@ -23,6 +23,21 @@ const mockFrameworkStatus = () => {
   };
 };
 
+const mockFailedFrameworkStatus = () => {
+  return {
```

Comment:
Make sure only mock job to fail when the first creation, i.e. not when we update it later. 
Otherwise, the mocked status is more complex (need status merge)